pkg/users/account-details.js

```diff
@@ -96,40 +96,50 @@
             .then(parse_expire);
 }
 
-export function AccountDetails({ accounts, groups, current_user, user, shells }) {
+export function AccountDetails({ account, groups, isLoading, current_user, shells }) {
     const [expiration, setExpiration] = useState(null);
 
+    const user = useMemo(() => account.name, [account.name]);
+    const [isLocked, setIsLocked] = useState(account.isLocked);
+
+    const [editedRealName, setEditedRealName] = useState(null);
+    const [comittingRealName, setCommittingRealName] = useState(false);
+
+    const [disableLockedEdit, setDisableLockedEdit] = useState(false);
+
     useEffect(() => {
         get_expire(user).then(setExpiration);
-    }, [user, accounts]);
+    }, [user]);
 
-    const [edited_real_name, set_edited_real_name] = useState(null);
-    const [committing_real_name, set_committing_real_name] = useState(false);
+    // Only update isLocked field if account changes value
+    useEffect(() => {
+      setIsLocked(account.isLocked);
+    }, [account.isLocked])
 
-    const [edited_locked, set_edited_locked] = useState(null);
 
-    function change_real_name() {
-        if (edited_real_name === null || edited_real_name === undefined)
+    function changeRealName() {
+        if (editedRealName === null || editedRealName === undefined)
             return;
 
-        set_committing_real_name(true);
+        setCommittingRealName(true);
 
         // TODO: unwanted chars check
-        cockpit.spawn(["/usr/sbin/usermod", user, "--comment", edited_real_name],
+        cockpit.spawn(["/usr/sbin/usermod", user, "--comment", editedRealName],
                       { superuser: "try", err: "message" })
                 .then(() => {
-                    set_edited_real_name(null);
-                    set_committing_real_name(false);
+                    setEditedRealName(null);
+                    setCommittingRealName(false);
                 })
                 .catch(error => {
-                    set_edited_real_name(null);
-                    set_committing_real_name(false);
+                    setEditedRealName(null);
+                    setCommittingRealName(false);
                     show_unexpected_error(error);
                 });
     }
 
     function change_locked(value, dont_retry_if_stuck) {
-        set_edited_locked(value);
+        setIsLocked(value);
+        setDisableLockedEdit(true);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-21865-5e987a2f-20250813-131607-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)